src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Unary.cs

```diff
@@ -333,22 +333,22 @@ private void EmitConvert(UnaryExpression node, CompilationFlags flags)
                         paramType = paramType.GetElementType();
                     }
 
-                    UnaryExpression e = Expression.Convert(
+                    node = Expression.Convert(
                         Expression.Call(
                             node.Method,
                             Expression.Convert(node.Operand, paramType)
                         ),
                         node.Type
                     );
-
-                    EmitConvert(e, flags);
```

Comment:
It's not possible that the new node being created would hit one of the various if/else branches that are now being skipped due to this change?